app/src/common/shared/com/igalia/wolvic/PlatformSpecificBehavior.java

```diff
@@ -0,0 +1,11 @@
+package com.igalia.wolvic;
+
+import android.content.Intent;
+
+/**
+ * Methods that will be implemented by the platform-specific activities.
+ */
+public interface PlatformSpecificBehavior {
+
+    boolean shouldOpenInKioskMode(Intent intent);
+}
```

Comment:
Yes, eventually we should rethink this part of the architecture.

Reply:
I was thinking about this, and I think there are many issues related to the current approach of using an interface. For example we're breaking other flavours' build because the method would be undefined. Also it seems weird to "force" `PlatformActivity`s to implement an interface.

Considering the current architecture in which `PlatformActivity` does not have a common ancestor for the different platforms I think that what we should do is to define `shouldOpenInKioskMode` in `VRBrowserActivity` directly and then implement it in terms of `DeviceType.isHVRBuild()` etc...

I think that alternative is better because it won't break other flavours, and although it does not have an OO design, it fits better IMHO with the current constraints.